### DISCUSSION TITLE
Download-artifact doesn't find existing artifact

### DISCUSSION THREAD

**Discussion by llrs** [community/community#26375] [Actions]
<p>I have a problem with a <a href="https://github.com/llrs/BOE_historico/actions?query=workflow%3Atwitter-hourly" rel="noopener nofollow ugc">scheduled job</a>.</p>
<p>I prepare some data, do some actions and save a portion of the data content as an artifact (actions/upload-artifact@v2).<br>
On the next run of the workflow I recover the data (actions/download-artifact@v2) and do something and again save a modified version of the data (actions/upload-artifact@v2).</p>
<p>The problem I’m having is that apparently I’m saving the modified version of the data correctly, but on the next run the download-artifact reports “Unable to find any artifacts for the associated workflow”. While I can see it on the artifacts of the workflow.</p>

**Comment by airtower-luna** [comment]
<p>The documentation on <a href="https://docs.github.com/en/free-pro-team@latest/actions/guides/storing-workflow-data-as-artifacts#downloading-artifacts-during-a-workflow-run">Downloading artifacts during a workflow run</a> says this:</p>
<blockquote>
<p>The <a href="https://github.com/actions/download-artifact"> <code>actions/download-artifact</code> </a> action can be used to download previously uploaded artifacts during a workflow run.</p>
<p><strong>Note:</strong> You can only download artifacts in a workflow that were uploaded during the same workflow run.</p>
</blockquote>
<p>So you can’t access artifacts from previous workflow runs using <code>actions/download-artifact</code>. You could check if the API lets you look at previous workflow runs and their artifacts.</p>

**Comment by llrs** [comment, marked as answer]
<p>Oh, I missed it from the documentation of the action itself. Thanks <a class="mention" href="/airtower-luna">@airtower-luna</a> .<br>
Will try accessing it through the API, which indeed allows to look and <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#get-an-artifact" rel="noopener nofollow ugc">get artifacts</a>.<br>
There is already an action that does what I wanted: <a href="https://github.com/marketplace/actions/download-workflow-artifact" rel="noopener nofollow ugc">download-workflow-artifact</a></p>
